prediction_market_agent/agents/microchain_agent/app.py

```diff
@@ -48,20 +48,38 @@
 ALLOW_STOP = False
 
 
-def run_agent(agent: Agent, iterations: int, model: str) -> None:
-    with openai_costs(model) as costs:
+def run_agent(agent: Agent, iterations: int) -> None:
+    maybe_initialize_long_term_memory()
+    with openai_costs(st.session_state.model) as costs:
         with st.spinner("Agent is running..."):
             for _ in range(iterations):
                 agent.run(iterations=1, resume=True)
         st.session_state.running_cost += costs.cost
 
 
-def execute_reasoning(agent: Agent, reasoning: str, model: str) -> None:
-    with openai_costs(model) as costs:
-        agent.execute_command(f'Reasoning("{reasoning}")')
-        display_new_history_callback(agent)  # Run manually after `execute_command`
+def on_click_run_agent() -> None:
+    maybe_initialize_agent(system_prompt, bootstrap)
```

Comment:
Shouldn't this be in ` run_agent` function as well?

Reply:
Not needed anymore, other branch was used